httpx/_transports/asgi.py

```diff
@@ -25,6 +25,76 @@ def create_event() -> "Event":
         return asyncio.Event()
 
 
+async def create_background_task(async_fn: typing.Callable) -> typing.Callable:
+    if sniffio.current_async_library() == "trio":
+        import trio
+
+        nursery_manager = trio.open_nursery()
+        nursery = await nursery_manager.__aenter__()
+        nursery.start_soon(async_fn)
+
+        async def aclose(exc: Exception = None) -> None:
+            if exc is not None:
+                await nursery_manager.__aexit__(type(exc), exc, exc.__traceback__)
+            else:
+                await nursery_manager.__aexit__(None, None, None)
+
+        return aclose
+
+    else:
+        import asyncio
+
+        loop = asyncio.get_event_loop()
+        task = loop.create_task(async_fn())
+
+        async def aclose(exc: Exception = None) -> None:
+            if not task.done():
+                task.cancel()
+
+        return aclose
+
+
+def create_channel(
+    capacity: int,
+) -> typing.Tuple[
+    typing.Callable[[], typing.Awaitable[bytes]],
+    typing.Callable[[bytes], typing.Awaitable[None]],
+]:
+    if sniffio.current_async_library() == "trio":
+        import trio
+
+        send_channel, receive_channel = trio.open_memory_channel[bytes](capacity)
+        return receive_channel.receive, send_channel.send
+
+    else:
+        import asyncio
+
+        queue: asyncio.Queue[bytes] = asyncio.Queue(capacity)
+        return queue.get, queue.put
+
+
+async def run_until_first_complete(*async_fns: typing.Callable) -> None:
+    if sniffio.current_async_library() == "trio":
+        import trio
+
+        async with trio.open_nursery() as nursery:
+
+            async def run(async_fn: typing.Callable) -> None:
+                await async_fn()
+                nursery.cancel_scope.cancel()
+
+            for async_fn in async_fns:
+                nursery.start_soon(run, async_fn)
+
+    else:
+        import asyncio
+
+        coros = [async_fn() for async_fn in async_fns]
+        done, pending = await asyncio.wait(coros, return_when=asyncio.FIRST_COMPLETED)
+        for task in pending:
+            task.cancel()
```

Comment:
While technically not required, I think it would be a good practice to await the cancelled tasks (in `suppress(CancelledError)` block), that way they wouldn't show with `PYTHONASYNCIODEBUG`.

Reply:
Thanks, I think I addressed this in the latest state of the PR. :+1: